src/store/books/details/BookDetailsContext.tsx

```diff
@@ -33,6 +42,38 @@ const BookDetailsContextProvider = ({ bookId, children }: Props) => {
     setLoading(false);
   };
 
+  const hideConfirmationModal = () => setShowedConfirmationModal(false);
+
+  const showSuccessScreen = () => {
+    setShowedSuccessMessage(true);
+    setShowedConfirmationModal(false);
+    setTimeout(() => setShowedSuccessMessage(false), 2000);
+  };
+
+  const sendRentalRequest = async () => {
+    await bookService.postRentalRequest({
+      bookId: bookId.toString(),
+      requestedStatus: bookStatus,
+    });
+  };
+
+  const performAction = () => {
```

Comment:
Let's actually have rentABook and returnABook methods instead of performAction. You can have internal "performBookAction(type: actionType)" that will be called by the rentABook or returnABook, so we reduce code duplication.